packages/fxa-auth-server/lib/payments/stripe.ts

```diff
@@ -815,6 +815,20 @@ export class StripeHelper {
     return !!subscription;
   }
 
+  /**
+   * Returns true if the Stripe customer has an active subscription.
+   */
+  async hasActiveSubscription(uid: string): Promise<Boolean> {
+    const customer = await this.fetchCustomer(uid, ['subscriptions']);
+    if (!customer) {
+      return false;
+    }
+    const subscription = customer.subscriptions?.data.find((sub) =>
+      ACTIVE_SUBSCRIPTION_STATUSES.includes(sub.status)
+    );
+    return !!subscription;
+  }
```

Comment:
This new helper looks suspiciously similar to `hasSubscriptionRequiringPaymentMethod`, except:
* It takes in a `uid`, since we don't know if there is a Stripe customer yet for this user.
* It doesn't care if the subscription is marked `cancel_at_period_end`.

If there's a way to make this DRYer, please let me know.